gradio/utils.py

```diff
@@ -225,13 +224,16 @@ def same_children_recursive(children1, chidren2, mapping):
     same_children_recursive(children1, children2, mapping)
 
     for d1, d2 in zip(config1["dependencies"], config2["dependencies"]):
-        for t1, t2 in zip(d1["targets"], d2["targets"]):
+        for t1, t2 in zip(d1.pop("targets"), d2.pop("targets")):
```

Comment:
I think it would be helpful to add a comment about why popping is necessary here.

Reply:
The reason is so that we can test the remaining items dictionaries for equivalence later on when we do: `assert d1 == d2`